File: src/mosquitto-1.4.5/misc/currentcost/cc128_read.pl
#!/usr/bin/perl -w

# Reads data from a Current Cost device via serial port.
# Spawns 

use strict;
use Device::SerialPort qw( :PARAM :STAT 0.07 );

my $pubclient = "mosquitto_pub -t sensors/cc128/raw -q 2 -l";
my $PORT = "/dev/ttyUSB0";
local $| = 1;

my $ob = Device::SerialPort->new($PORT);
$ob->baudrate(57600);
$ob->write_settings;

open(SERIAL, "+<$PORT");
open(MQTT, "|$pubclient");
while (my $line = <SERIAL>) {
	print(MQTT "$line");
}	    

close(MQTT);

File: src/mosquitto-1.4.5/misc/currentcost/cc128_read.py
#!/usr/bin/python -u

import mosquitto
import serial

usb = serial.Serial(port='/dev/ttyUSB0', baudrate=57600)

mosq = mosquitto.Mosquitto()
mosq.connect("localhost")
mosq.loop_start()

running = True
try:
    while running:
        line = usb.readline()
        mosq.publish("cc128/raw", line)
except usb.SerialException, e:
    running = False

mosq.disconnect()
mosq.loop_stop()

